services/email-service/controllers/firestoreController.js

```
@@ -0,0 +1,92 @@
+                });
+                console.log("Connected to Firebase Firestore locally with firebase configuration file");
+            }
+            else {
+                firebaseApp = initializeApp();
+                console.log("Connected to Firebase Firestore over cloud configuration");
+            }
+        }
+        catch (error) {
+            console.log("\x1b[31m", "ERROR: Unable to connect to Firestore Instance\nAre you running locally or over the cloud?");
+            throw error;
+        }
+
+        // https://stackoverflow.com/questions/77061688/how-can-i-connect-to-a-non-default-firestore-database-using-node-using-multiple
+        // try connecting to the firestore database defined in the environment variables
+        try {
+            firestoreDatabase = getFirestore(firebaseApp, firestoreId);
+            console.log(`Connected to Firestore Database: ${firestoreId}`);
+        }
+        catch (error) {
+            console.log("\x1b[31m", "ERROR: Unable to connect to Firestore Database\nIs the Firestore Database ID correct?");
+            throw error;
+        }
+    },
+
+    registerEmail: async (email, code) => {
+        try {
+            const docRef = collection(firestoreDatabase, collectionName, email);
+            await setDoc(docRef, { code: code });
+        }
+        catch (error) {
+            console.log("\x1b[31m", "ERROR: Unable to register email in Firestore Database");
+            throw error;
+        }
+    },
+
+    authorizeEmail: async (email, authCode) => {
+        try {
+            const docRef = collection(firestoreDatabase, collectionName, email);
+            const doc = await getDoc(docRef);
+
+            if (doc.exists()) {
+                const data = doc.data();
+                if (data.code === authCode) {
+                    return true;
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
```

if receiving invalid email, should throw instead of returning false

---

throws error rather than returning false now